### PR TITLE
Improve wallet-accounts test

### PR DESCRIPTION
**Pull request by ryanofsky**
Add comments and

- Verify sending to a account causes getaccountaddress to generate new addresses.
- Verify sending to a account causes getreceivedbyaccount to return amount received.
- Verify ways setaccount updates the accounts of existing addresses.

### CODE REVIEW

File: test/functional/wallet-accounts.py [hunk 1]
```diff
@@ -72,62 +72,135 @@ def run_test(self):
...
+    old_account.addresses.remove(address)
+    new_account.add_address(address)
+
+    # Calling setaccount on an address which was previously the receiving
+    # address of a different account should reset the receiving address of
+    # the old account, causing getaccountaddress to return a brand new
+    # address.
+    if address == old_account.receive_address:
+        new_address = node.getaccountaddress(old_account.name)
+        assert_equal(new_address not in old_account.addresses, True)
```
Comment (promag) [inline test/functional/wallet-accounts.py]: `assert ... not in ...;`?

**Review by promag — COMMENTED** [review]
Is it worth since accounts is deprecated?

**Comment by ryanofsky** [comment]
> Is it worth since accounts is deprecated?

Yes, all of the new tests apply to both accounts and labels. The new tests helped me identify two bugs in #7729.

File: test/functional/wallet-accounts.py (hunk 1 above)
Comment (ryanofsky) [reply test/functional/wallet-accounts.py]: > assert ... not in ...;?

Assert statements are skipped when python is run in optimized mode. I only use assert statements in tests when asserting conditions that indicate bugs in test setup, as opposed to conditions that could be caused by bugs in the code under test. (If you're familiar with google test framework, this is the distinction between CHECK and ASSERT_TRUE).

Comment (promag) [reply test/functional/wallet-accounts.py]: Thanks for the explanation, wasn't aware of this.

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/11552/commits/bc9c0a7b74eab853fc21402d62a30dacd77a8843

**Comment by MarcoFalke** [comment]
post merge utACK bc9c0a7

**Pull request merged**